components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -0,0 +1,2222 @@
+         */
+        } else {
+            Assert.assertEquals(remoteFileGroupId, lightAdmin.groupId);
+            Assert.assertEquals(imageGroupId, lightAdmin.groupId);
+            Assert.assertEquals(datasetGroupId, lightAdmin.groupId);
+            Assert.assertEquals(datasetImageLinkGroupId, lightAdmin.groupId);
+        }
+        /* now, having moved the dataset, image, original file and link in the group of normalUser,
+         * try to change the ownership of the dataset to the normalUser */
+        /* Chowning the dataset should fail in case you have not both of
+         * isAdmin & Chown permissions which are
+         * captured in the boolean importYourGroupAndChgrpAndChownExpectSuccess.
+         * Additionally, in this boolean is permChgrp, which was necessary for the
+         * previous step of moving the data into normalUser's group.
+         * A successful chowning of the dataset will chown the linked image
+         * and the link too.*/
+        if (importYourGroupAndChgrpAndChownExpectSuccess) {/* whole workflow2 succeeded */
+            doChange(client, factory, Requests.chown().target(sentDat).toUser(normalUser.userId).build(), true);
+            remoteFileGroupId = ((RLong) iQuery.projection(
+                    "SELECT details.group.id FROM OriginalFile o WHERE o.id = :id",
+                    new ParametersI().addId(remoteFile.getId())).get(0).get(0)).getValue();
+            imageGroupId = ((RLong) iQuery.projection(
+                    "SELECT details.group.id FROM Image i WHERE i.id = :id",
+                    new ParametersI().addId(image.getId())).get(0).get(0)).getValue();
+            datasetGroupId = ((RLong) iQuery.projection(
+                    "SELECT details.group.id FROM Dataset d WHERE d.id = :id",
+                    new ParametersI().addId(sentDat.getId())).get(0).get(0)).getValue();
+            long datasetImageLinkId = ((RLong) iQuery.projection(
+                    "SELECT id FROM DatasetImageLink WHERE parent.id = :id",
+                    new ParametersI().addId(sentDat.getId())).get(0).get(0)).getValue();
+            datasetImageLinkGroupId = ((RLong) iQuery.projection(
+                    "SELECT details.group.id FROM DatasetImageLink WHERE parent.id = :id",
+                    new ParametersI().addId(sentDat.getId())).get(0).get(0)).getValue();
+            /* image, dataset and link are in the normalUser's group and belong to normalUser */
+            assertOwnedBy(remoteFile, normalUser);
+            Assert.assertEquals(remoteFileGroupId, normalUser.groupId);
+            assertOwnedBy(image, normalUser);
+            Assert.assertEquals(imageGroupId, normalUser.groupId);
+            assertOwnedBy(sentDat, normalUser);
+            Assert.assertEquals(datasetGroupId, normalUser.groupId);
+            assertOwnedBy((new DatasetImageLinkI (datasetImageLinkId, false)), normalUser);
+            Assert.assertEquals(datasetImageLinkGroupId, normalUser.groupId);
+        } els
```

Comment:
fixed in https://github.com/openmicroscopy/openmicroscopy/pull/5159/commits/3917d9c251f02fdee0f88fd09eb6a6e08928ac81